core-processor/src/ext.rs

```diff
@@ -689,6 +685,43 @@ impl EnvExt for Ext {
     fn forbidden_funcs(&self) -> &BTreeSet<&'static str> {
         &self.context.forbidden_funcs
     }
+
+    fn counters(&self) -> (u64, u64) {
+        (
+            self.context.gas_counter.left(),
+            self.context.gas_allowance_counter.left(),
+        )
+    }
+
+    fn update_counters(&mut self, gas: u64, allowance: u64) {
+        let gas_left = self.context.gas_counter.left();
+        if gas_left > gas {
+            self.context.gas_counter.charge(gas_left - gas);
+        } else {
+            self.context.gas_counter.refund(gas - gas_left);
+        }
```

Comment:
probably check for equality and do nothing?

Reply:
I would leave it as is